api/envoy/api/v2/core/protocol.proto

```diff
@@ -83,6 +84,17 @@ message Http1ProtocolOptions {
   // Describes how the keys for response headers should be formatted. By default, all header keys
   // are lower cased.
   HeaderKeyFormat header_key_format = 4;
+
+  // Enables trailers for HTTP1. By default the HTTP/1 codec drops proxied trailers.
+  //
+  // .. attention::
+  //
+  //   Note that this only happens when Envoy is chunk encoding which occurs when:
+  //   - The request is Http1.1
```

Comment:
Are you asking how is different if you apply it to the downstream versus upstream? You can apply the enable_trailers either on the upstream or downstream. So if you enable_trailers on only one side, it will enable decoding and encoding trailers for that one side.

If you know your downstream is going to be H1 and upstream will always be H2, then you only have to apply enable_trailers on the downstream since H2 will always encode/decode already. 

Not sure if that answered your question or not.